elements_imaging/imaging.py

```diff
@@ -152,63 +151,74 @@ class Processing(dj.Computed):
+            elif method == 'caiman':
+                loaded_caiman = loaded_result
+                key = {**key, 'processing_time': loaded_caiman.creation_time}
+            else:
+                raise NotImplementedError('Unknown method: {}'.format(method))
+        elif task_mode == 'trigger':
+            raise NotImplementedError(f'Automatic triggering of {method} analysis is not yet supported')
+        else:
+            raise ValueError(f'Unknown task mode: {task_mode}')
 
-                data_dir = pathlib.Path(get_suite2p_dir(key))
-                loaded_suite2p = suite2p_loader.Suite2p(data_dir)
-                key = {**key, 'proc_completion_time': loaded_suite2p.creation_time, 'proc_curation_time': loaded_suite2p.curation_time}
-                # Insert file(s)
-                root = pathlib.Path(scan.get_imaging_root_data_dir())
-                output_files = data_dir.glob('*')
-                output_files = [f.relative_to(root).as_posix() for f in output_files if f.is_file()]
+        self.insert1(key)
 
-            elif method == 'caiman':
-                from .readers import caiman_loader
 
-                data_dir = pathlib.Path(get_caiman_dir(key))
-                loaded_caiman = caiman_loader.CaImAn(data_dir)
+@schema
+class Curation(dj.Manual):
+    definition = """
+    -> Processing
+    curation_id: int
+    ---
+    curation_time: datetime             # time of generation of this set of curated results 
+    curation_output_dir: varchar(255)   # output directory of the curated results, relative to root data directory
+    manual_curation: bool               # has manual curation been performed on this result?
+    curation_note='': varchar(2000)  
+    """
 
-                key = {**key, 'proc_completion_time': loaded_caiman.creation_time,
-                              'proc_curation_time': loaded_caiman.curation_time}
+    def create1_from_processing_task(self, key, is_curated=False, curation_note=''):
+        """
+        A convenient function to create a new corresponding "Curation" for a particular "ClusteringTask"
+        """
+        if key not in Processing():
+            raise ValueError(f'No corresponding entry in Clustering available for: {key}; do `Processing.populate(key)`')
 
-                # Insert file(s)
-                root = pathlib.Path(scan.get_imaging_root_data_dir())
-                output_files = [loaded_caiman.caiman_fp.relative_to(root).as_posix()]
-            else:
-                raise NotImplementedError('Unknown method: {}'.format(method))
+        output_dir = (ProcessingTask & key).fetch1('processing_output_dir')
+        method, loaded_result = get_loader_result(key, ProcessingTask)
 
-            self.insert1(key)
-            self.ProcessingOutputFile.insert([{**key, 'file_path': f} for f in output_files], ignore_extra_fields=True)
+        if method == 'suite2p':
+            loaded_suite2p = loaded_result
+            curation_time = loaded_suite2p.creation_time
```

Comment:
Since we are now only using `creation_time`, should we remove `curation_time` from `elements_imaging/readers/caiman_loader.py` and `elements_imaging/readers/suite2p_loader.py`?

Reply:
we may use `curation_time` in the future, as a way of detecting latest curation from a set of results. Let's keep it for now